File: AtCoder/AtCoder Beginner Contest/086/C/Go/ABC_086_C_01.go
package main

import (
	"fmt"
)

func abs(v int) int {
	if v > 0 {
		return v
	} else {
		return -v
	}
}

func main() {
	var N int
	fmt.Scanf("%d", &N)

	var (
		t_old int = 0
		x_old int = 0
		y_old int = 0
	)

	var t, x, y int
	var time, dist int
	for i := 0; i < N; i++ {
		fmt.Scanf("%d %d %d", &t, &x, &y)

		time = abs(t_old - t)
		dist = abs(x_old - x) + abs(y_old - y)

		if ((t + x + y) % 2 != 0 || time < dist) {
			fmt.Printf("No")
			return
		}

		t_old, x_old, y_old = t, x, y
	}
	fmt.Printf("Yes")
}

File: AtCoder/AtCoder Beginner Contest/086/C/C/ABC_086_C_01.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int N;
    scanf("%d", &N);

    int t_old = 0, x_old = 0, y_old = 0;
    int time, dist;
    for (int i = 0; i < N; i++) {
        int t, x, y;
        scanf("%d %d %d", &t, &x, &y);

        time = abs(t_old - t);
        dist = abs(x_old - x) + abs(y_old - y);

        if ((t + x + y) % 2 != 0 || time < dist) {
            printf("No");
            return 0;
        }

        t_old = t, x_old = x, y_old = y;
    }
    printf("Yes");
    return 0;
}